src/net_processing.cpp

```diff
@@ -2304,8 +2370,7 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 // Never relay transactions that we would assign a non-zero DoS
                 // score for, as we expect peers to do the same with us in that
                 // case.
-                int nDoS = 0;
-                if (!state.IsInvalid(nDoS) || nDoS == 0) {
+                if (!state.IsInvalid() || !MayResultInDisconnect(state, false)) {
```

Comment:
At this point, it's clearer to switch the ordering of the conditionals:

```
                if (state.IsInvalid() && MayResultInDisconnect(state, false)) {
                    LogPrintf("Not relaying invalid transaction %s from whitelisted peer=%d (%s)\n", tx.GetHash().ToString(), pfrom->GetId(), FormatStateMessage(state));
                } else {
                    LogPrintf("Force relaying tx %s from whitelisted peer=%d\n", tx.GetHash().ToString(), pfrom->GetId());
                    RelayTransaction(tx, connman);
                }
```

if (THING AND THING) is more natural than if (^THING OR ^THING)

Reply:
Done.